src/mongo_logger.ts

```diff
@@ -728,6 +728,10 @@ export class MongoLogger {
   logDestination: MongoDBLogWritable;
   logDestinationIsStdErr: boolean;
   pendingLog: PromiseLike<unknown> | unknown = null;
+  willLog: Record<MongoLoggableComponent, Record<SeverityLevel, boolean>> = {} as Record<
```

Comment:
Can we change a type here in a way that we don't need a casting?

Reply:
What do you suggest?

Reply:
I tried this. While it gets rid of the type casting being necessary its also not an accurate type for our willLog object as it doesn't assert all the `MongoLoggableComponent`s will be in the object. I've found a workaround though, if you see my most recent push!